x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/chart_switch/chart_option_append.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import './chart_switch.scss';
+import React, { ReactNode } from 'react';
+import { EuiFlexItem, EuiIconTip, EuiBetaBadge, EuiFlexGroup } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+
+export const getDataLossWarning = (dataLoss: 'nothing' | 'layers' | 'everything' | 'columns') => {
+  if (dataLoss === 'nothing') {
+    return;
+  }
+  if (dataLoss === 'everything') {
+    return i18n.translate('xpack.lens.chartSwitch.dataLossEverything', {
+      defaultMessage: 'Changing to this visualization clears the current configuration.',
+    });
+  }
+  if (dataLoss === 'layers') {
+    return i18n.translate('xpack.lens.chartSwitch.dataLossLayersDescription', {
+      defaultMessage:
+        'Changing to this visualization modifies currently selected layer`s configuration and removes all other layers.',
+    });
+  } else
+    return i18n.translate('xpack.lens.chartSwitch.dataLossColumns', {
+      defaultMessage: `Changing to this visualization modifies the current configuration.`,
+    });
+};
+
+const DataLossWarning = ({ content, id }: { content: ReactNode; id: string }) => {
+  if (!content) return null;
+  return (
+    <EuiFlexItem grow={false}>
+      <EuiIconTip
+        size="s"
+        aria-label={i18n.translate('xpack.lens.chartSwitch.dataLossLabel', {
+          defaultMessage: 'Warning',
+        })}
+        type="dot"
+        color="warning"
+        content={content}
+        iconProps={{
+          className: 'lnsChartSwitch__chartIcon',
+          'data-test-subj': `lnsChartSwitchPopoverAlert_${id}`,
+        }}
+      />
+    </EuiFlexItem>
+  );
+};
+
+export const ExperimentalBadge = () => {
+  return (
+    <EuiFlexItem grow={false}>
+      <EuiBetaBadge
+        css={css`
+          vertical-align: middle;
+        `}
+        iconType="beaker"
+        label={i18n.translate('xpack.lens.chartSwitch.experimentalLabel', {
+          defaultMessage: 'Technical preview',
+        })}
+        size="s"
+      />
```

Comment:
Would it make sense to also apply a tooltip to this badge that states "Technical preview", similar to how we do with the same badge in layer settings?

![CleanShot 2024-03-14 at 16 14 36](https://github.com/elastic/kibana/assets/3884767/c9b08ba5-35cd-44c9-9d74-27983914d180)

Reply:
Sure!

Reply:
and done!